packages/core-data/src/hooks/use-entity-record.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { IDLE, SUCCESS, ERROR, RESOLVING } from './constants';
+
+/**
+ * @typedef {IDLE|RESOLVING|SUCCESS|ERROR} EntityRecordStatus
+ */
+
+/**
+ * @typedef {Object} EntityRecordResolution
+ * @property {Object}             record       the requested entity record
+ * @property {Object}             editedRecord the requested entity record with any edits applied
+ * @property {boolean}            isMissing    is the record missing after the resolver has finished?
+ * @property {boolean}            isResolving  is the record still being resolved?
+ * @property {boolean}            hasResolved  is the record resolved by now?
+ * @property {boolean}            hasEdits     were there eny edits applied to this entity record?
+ * @property {EntityRecordStatus} status       resolution status
+ */
+
+/**
+ * Resolves the specified entity record.
+ *
+ * @param {string} kind     Kind of the deleted entity.
+ * @param {string} name     Name of the deleted entity.
+ * @param {string} recordId Record ID of the deleted entity.
+ *
+ * @example
+ * ```js
+ * import { useEntityRecord } from '@wordpress/core-data';
+ *
+ * function HammerPriceDisplay( { hammerId } ) {
+ *   const { record, isResolving } = useEntityRecord( 'my-shop', 'hammer', hammerId );
+ *
+ *   if ( isResolving ) {
+ *     return 'Loading...';
+ *   }
+ *
+ *   return new Intl.NumberFormat( 'en-US', {
+ *     style: 'USD',
+ *     currency,
+ *   } ).format( record.price );
```

Comment:
I know it's just an example but this code could be improved in performance. Newing a `NumberFormat` instance on every render is costly, we can move it outside the Component to only initialize it once. (Also seems like `currency` is missing in the scope?)

We might want to be careful when writing examples like this since they're often the first piece of code new contributors see and copy-paste from 😆 .